implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -208,7 +208,10 @@ private Object findClaimValue(String claimPath, Map<String, Object> claimsMap, S
 
     protected VerificationKeyResolver getKeyResolver(JWTAuthContextInfo authContextInfo) throws UnresolvableKeyException {
         if (keyResolver == null) {
-            keyResolver = new KeyLocationResolver(authContextInfo);
+            synchronized (this) {
```

Comment:
But it is `volatile`, the very worst thing that can happen it can be created more than once but ultimately there will be only a single instance used

Reply:
Never mind, guess this pattern is better anyway